tests/test_flight.py

```diff
@@ -849,17 +868,62 @@ def test_latlon_conversions2(mock_show):
     )
 
     test_env = Environment(
-        railLength=5,
         latitude=0,
         longitude=0,
         elevation=1400,
     )
 
     test_flight = Flight(
-        rocket=test_rocket, environment=test_env, inclination=85, heading=0
+        rocket=test_rocket,
+        environment=test_env,
+        railLength=5,
+        inclination=85,
+        heading=0,
     )
 
     test_flight.postProcess()
 
     assert abs(test_flight.longitude(test_flight.tFinal) - 0) < 1e-12
     assert test_flight.latitude(test_flight.tFinal) > 0
+
+
+@pytest.mark.parametrize(
+    "rail_length, out_of_rail_time",
+    [
+        (0.52, 0.5180212542878443),
+        (5.2, 5.180378138072207),
+        (50.2, 50.00897551720473),
+        (100000, 100003.35594050681),
+    ],
+)
+def test_rail_length(rocket, example_env, rail_length, out_of_rail_time):
+    """Test the rail length parameter of the Flight class. This test simply
+    simulate the flight using different rail lengths and checks if the expected
+    out of rail altitude is achieved. Four different rail lengths are
+    tested: 0.001, 1, 10, and 100000 meters. This provides a good test range.
+    Currently, if a rail length of 0 is used, the simulation will fail in a
+    ZeroDivisionError, which is not being tested here.
+
+    Parameters
+    ----------
+    rocket : rocketpy.Rocket
+        The rocket to be simulated. In this case, the fixture rocket is used.
+        See the conftest.py file for more information.
+    example_env : rocketpy.Environment
+        The environment to be simulated. In this case, the fixture environment
+        is used. See the conftest.py file for more information.
+    rail_length : float, int
+        The length of the rail in meters. It must be a positive number. See the
+        Flight class documentation for more information.
+    out_of_rail_time : float, int
+        The expected time at which the rocket leaves the rail in seconds.
+    """
+    test_flight = Flight(
+        rocket=rocket,
+        environment=example_env,
+        railLength=rail_length,
+        inclination=85,
+        heading=0,
+        terminateOnApogee=True,
+    )
+    assert (test_flight.z(test_flight.outOfRailTime) - out_of_rail_time) < 1e-6
```

Comment:
Am I just extremely tired here, or this testing is asserting that the altitude at lift off is lower than the out of rail time?
This is non-sense.

Reply:
we should either use abs(a - b) < 1e-6 or other approx. function available in pytest or numpy.